presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/QueryBuilder.java

```diff
@@ -169,11 +220,12 @@ private String quote(String name)
         return quote + name + quote;
     }
 
-    private static String encode(Object value)
+    private static String bindValue(Object value, Type type, List<TypeAndValue> accumulator)
     {
-        if (value instanceof Number || value instanceof Boolean) {
-            return value.toString();
+        if (isAcceptedType(type)) {
+            accumulator.add(new TypeAndValue(type, value));
+            return "?";
         }
-        throw new UnsupportedOperationException("Can't handle type: " + value.getClass().getName());
+        throw new UnsupportedOperationException("Can't handle type: " + type.getClass().getName());
```

Comment:
you should just do this at the top
checkArgument(isAcceptedType(type), "Can't handle type: %s" type)